File: Automation_Testing/Web_Selenium_Tests/pages/doctorDashboardPage.js
const { By, until } = require('selenium-webdriver');
const config = require('../config');

class DoctorDashboardPage {
    constructor(driver) {
        this.driver = driver;
        this.screenDoctorDashboard = By.id('screen-dashboard-doctor');
        this.welcomeMsg = By.id('doctor-welcome-msg');
        this.doctorIdDisplay = By.id('doctor-id-display');
        this.statPatients = By.id('stat-patients');
        this.statAppts = By.id('stat-appts');
        this.statApprovals = By.id('stat-approvals');
        this.statReviews = By.id('stat-reviews');
        this.btnSync = By.xpath("//button[contains(text(), 'SYNC DATA')]");
        this.btnScheduleVisit = By.xpath("//button[contains(text(), 'Schedule New Visit')]");
        this.modalAddAppt = By.id('modal-add-appt');
    }

    async isDashboardDisplayed() {
        const el = await this.driver.wait(until.elementLocated(this.screenDoctorDashboard), config.EXPLICIT_WAIT);
        return await el.isDisplayed();
    }

    async openScheduleVisitModal() {
        const btn = await this.driver.findElement(this.btnScheduleVisit);
        await btn.click();
        await this.driver.wait(until.elementIsVisible(await this.driver.findElement(this.modalAddAppt)), config.EXPLICIT_WAIT);
    }
}

module.exports = DoctorDashboardPage;

File: automation/pages/doctor_dashboard_page.py
from selenium.webdriver.common.by import By
from automation.pages.base_page import BasePage

class DoctorDashboardPage(BasePage):
    WELCOME_MSG = (By.ID, "doctor-welcome-msg")
    STAT_PATIENTS = (By.ID, "stat-patients")
    STAT_APPTS = (By.ID, "stat-appts")
    STAT_APPROVALS = (By.ID, "stat-approvals")
    STAT_REVIEWS = (By.ID, "stat-reviews")
    NEW_VISIT_BTN = (By.XPATH, "//button[contains(text(), '+ NEW VISIT')]")
    LOGOUT_BTN = (By.XPATH, "//button[contains(text(), '🚪')]")

    def is_dashboard_visible(self):
        return self.is_displayed(By.ID, "screen-dashboard-doctor")

    def get_welcome_text(self):
        return self.get_text(*self.WELCOME_MSG)

    def logout(self):
        self.click(*self.LOGOUT_BTN)
